tests/configs/signing-txns-with-webauthn.ts

```diff
@@ -0,0 +1,378 @@
+    action: 'clickButtonByText',
+    buttonText: 'hardhat.config.ts',
+  },
+  'hardhat-config': {
+    action: 'writeToFile',
+    filepath: 'tests-output/zksync-webauthn/contracts/hardhat.config.ts',
+  },
+  'make-deploy-script': {
+    action: 'runCommand',
+    commandFolder: 'tests-output/zksync-webauthn/contracts',
+  },
+  'open-deploy-script': {
+    action: 'clickButtonByText',
+    buttonText: 'deploy.ts',
+  },
+  'deploy-script': {
+    action: 'writeToFile',
+    filepath: 'tests-output/zksync-webauthn/contracts/deploy/deploy.ts',
+  },
+  'compile-and-deploy': {
+    action: 'runCommand',
+    commandFolder: 'tests-output/zksync-webauthn/contracts',
+    saveOutput: 'tests-output/zksync-webauthn/contracts/deploy-output.txt',
+  },
+};
+
+const frontendPart1Steps: IStepConfig = {
+  'init-next-app': {
+    action: 'runCommand',
+    commandFolder: 'tests-output/zksync-webauthn',
+    preCommand: '<COMMAND> --ts --eslint --src-dir --use-npm --no-app --no-tailwind --import-alias "@/*" frontend',
+  },
+  'wait-for-nextjs-init': {
+    action: 'wait',
+    timeout: 5000,
+  },
+  'install-nextjs-deps': {
+    action: 'runCommand',
+    commandFolder: 'tests-output/zksync-webauthn',
+  },
+  'wait-for-nextjs-deps': {
+    action: 'wait',
+    timeout: 5000,
+  },
+  'open-home-page': {
+    action: 'clickButtonByText',
+    buttonText: 'index.tsx',
+  },
+  'home-page-code': {
+    action: 'writeToFile',
+    filepath: 'tests-output/zksync-webauthn/frontend/src/pages/index.tsx',
+  },
+  'create-env-file': {
+    action: 'runCommand',
+    commandFolder: 'tests-output/zksync-webauthn/frontend',
+    useSetCommand: 'touch .env.local',
+  },
+  'extract-aa-factory-address': {
+    action: 'extractDataToEnv',
+    dataFilepath: 'tests-output/zksync-webauthn/contracts/deploy-output.txt',
+    envFilepath: 'tests-output/zksync-webauthn/frontend/.env.local',
+    variableName: 'NEXT_PUBLIC_AA_FACTORY_ADDRESS',
+    regex: /(?<=factory address:\s*)0x[a-fA-F0-9]{40}/i,
+  },
+  'extract-nft-contract-address': {
+    action: 'extractDataToEnv',
+    dataFilepath: 'tests-output/zksync-webauthn/contracts/deploy-output.txt',
+    envFilepath: 'tests-output/zksync-webauthn/frontend/.env.local',
+    variableName: 'NEXT_PUBLIC_NFT_CONTRACT_ADDRESS',
+    regex: /(?<=NFT CONTRACT ADDRESS:\s*)0x[a-fA-F0-9]{40}/i,
+  },
+  'extract-paymaster-contract-address': {
+    action: 'extractDataToEnv',
+    dataFilepath: 'tests-output/zksync-webauthn/contracts/deploy-output.txt',
+    envFilepath: 'tests-output/zksync-webauthn/frontend/.env.local',
+    variableName: 'NEXT_PUBLIC_PAYMASTER_ADDRESS',
+    regex: /(?<=PAYMASTER CONTRACT ADDRESS:\s*)0x[a-fA-F0-9]{40}/i,
+  },
+  'replace-deployed-contract-address': {
+    action: 'modifyFile',
+    filepath: 'tests-output/zksync-webauthn/frontend/src/pages/index.tsx',
+    atLine: 5,
+    removeLines: [5, 7, 9],
+    useSetData: `export const AA_FACTORY_ADDRESS = process.env.NEXT_PUBLIC_AA_FACTORY_ADDRESS || '';
```

Comment:
Is this because it's harder to set env variables in tests than to add code?

Reply:
this is just so I can decouple the steps of replacing a variable and getting the variable from the output of another script